doc/manual/calling-c-and-fortran-code.rst

```diff
@@ -144,248 +149,554 @@ memory to be passed to the callee and filled in. Allocation of memory
+                                                                defined by-value, and is stored without a type-tag
+
+``type ...; end``               ``nothing``                     "Singleton" :: a Leaf Type or Immutable with no fields
+
+``(...)`` or ``tuple(...)```    ``(1,2,3)``                     "Tuple" :: an immutable data-structure similar to an
+                                                                anonymous immutable type, or a constant array. it's
+                                                                storage semantics are TBD.
+
+``typealias``                   Not applicable here             Type aliases, and other similar mechanisms of
+                                                                doing type indirection, are resolved to their base
+                                                                type (this includes assigning a type to another name,
+                                                                or getting the type out of a function call)
+==============================  ==============================  ======================================================
+
+Bits Types:
+~~~~~~~~~~~
+
+There are several special types to be aware of, as no other type can be defined to behave the same:
+
+:Float32: Exactly corresponds to the ``float`` type in C (or ``REAL*4`` in Fortran)
+:Float64: Exactly corresponds to the ``double`` type in C (or ``REAL*8`` in Fortran)
+:Complex64: Exactly corresponds to the ``complex float`` type in C (or ``COMPLEX*8`` in Fortran)
+:Complex128: Exactly corresponds to the ``complex double`` type in C (or ``COMPLEX*16`` in Fortran)
+:Signed: Exactly corresponds to the ``signed`` type annotation in C (or any ``INTEGER`` type in Fortran). Any Julia type that is not a subtype of ``Signed`` is assumed to be unsigned
+:Ref{T}: Behaves like a ``Ptr{T}`` that owns it's memory
+:Array{T,N}:
+    When an array is passed to C as a ``Ptr{T}`` argument, it is
+    not reinterpret-cast: Julia requires that the element type of the
+    array matches ``T``, and then address of the first element is passed.
+
+    Therefore, if an ``Array`` contains data in the wrong format, it will
+    have to be explicitly converted using a call such as ``int32(a)``.
+
+    To pass an array ``A`` as a pointer of a different type *without*
+    converting the data beforehand (for example, to pass a ``Float64`` array
+    to a function that operates on uninterpreted bytes), you can either
+    declare the argument as ``Ptr{Void}`` or you can explicitly call
+    ``pointer(A)``.
+
+    If an array of eltype ``Ptr{T}`` is passed as a ``Ptr{Ptr{T}}`` argument, the Julia base library
+    `cconvert_gcroot` function will attempt to first make a null-terminated copy of the array with
+    each element replaced by it's ``cconvert`` version. This allows, for example, passing an ``argv``
+    pointer array of type ``Vector{ByteString}`` to an argument of type ``Ptr{Ptr{Cchar}}``.
 
```

Comment:
The sentence begins "If an array of eltype `Ptr{T}` is passed as a `Ptr{Ptr{T}}` argument".  Grammatically, that means that the business about making the copy only applies to this case.   Hence, it does not apply to passing "``Vector{ByteString}`` to an argument of type ``Ptr{Ptr{Cchar}}``".

Maybe you meant to write: "If an array whose eltype can be converted to `Ptr{T}` is passed as a `Ptr{Ptr{T}}` argument" ... ?